### PR TITLE
Database: Added the ability to insert dummy species data, and get the species lists.

### PR DESCRIPTION
**Pull request by MrDrummer**

### CODE REVIEW

**Review by CmdrGree — COMMENTED** [review]
I don't know what any of this does! :crab:

File: central_web_server/handle_database.js [hunk 1]
```diff
@@ -1,19 +1,137 @@
...
 
-var addSurveyResults = async function(results) {
-	let connection = mysql.createConnection(config.connection);
-	connection.connect();
-	
-	let queries = [];
-	for (let item of results) {
-		queries.push( "(" + connection.escape(item) + ")" );
+let exampleData = [
+	{
```
Comment (byran) [inline central_web_server/handle_database.js]: Move this to another file, do not put test data into production source code.

File: central_web_server/handle_database.js [hunk 2]
```diff
@@ -1,19 +1,137 @@
...
+		{"id":7, "name": "£1"}
+		]
 	}
-	
-	let response = await connection.query('INSERT INTO survey_results (species) VALUES ' + queries.join(","));
-	//console.log(response);
-	connection.end();
+];
+
+async function addDummyData(data) {
```
Comment (byran) [inline central_web_server/handle_database.js]: This function name seems odd, it's not adding dummy data, it's inserting valid data.

File: central_web_server/handle_database.js [hunk 3]
```diff
@@ -1,19 +1,137 @@
...
+  let groupEntryQuery = squel
+    .insert()
+    .into("species_group_entry")
+    .setFieldsRows([].concat.apply([], groupEntryData))
+    .toString()
+  console.log(groupEntryQuery);
+  let groupEntryId = await connection.query(groupEntryQuery);
+  console.log(groupEntryId);
+}
+function concatIds(groupId, startingIndex, qty, fieldNames, retStr = false) {
```
Comment (byran) [inline central_web_server/handle_database.js]: Add a new line between the end of the previous function and this one.

File: central_web_server/handle_database.js [hunk 4]
```diff
@@ -1,19 +1,137 @@
...
+  for (let id = startingIndex; id < (startingIndex + qty); id++) {
+    let obj = {};
+    obj[fieldNames[0]] = (retStr ? groupId.toString() : groupId);
+    obj[fieldNames[1]] = (retStr ? id.toString() : id);
+    output.push(obj);
+  }
+  return output;
+}
+
+//addDummyData(exampleData);
```
Comment (byran) [inline central_web_server/handle_database.js]: Remove this line

File: central_web_server/handle_database.js [hunk 5]
```diff
@@ -1,19 +1,137 @@
...
+  for (let group of Object.keys(structureObj)) {
+    output.push(structureObj[group]);
+  }
+  return output;
+}
+
+async function addSurveyResults() {
+  
 }
+getSpeciesLists();
```
Comment (byran) [inline central_web_server/handle_database.js]: I don't think you want this, remove this line.

File: central_web_server/handle_database.js [hunk 6]
```diff
@@ -1,19 +1,137 @@
...
+  for (let group of data) {
+    let insertData = [];
+    
+    let queryGroup = squel
+      .insert()
+      .into("species_group")
+      .setFieldsRows([{name: group.species_list}])
+      .toString()
+    //console.log(queryGroup);
+    let groupId = await connection.query(queryGroup);
```
Comment (byran) [inline central_web_server/handle_database.js]: The result isn't a groupId, it's the result of the query so name the variable to reflect this.

File: central_web_server/handle_database.js [hunk 7]
```diff
@@ -1,19 +1,137 @@
...
+    for (let species of group.species) {
+      insertData.push({name: species.name});
+    }
+    let querySpecies = squel
+      .insert()
+      .into("species")
+      .setFieldsRows(insertData)
+      .toString()
+    //console.log(querySpecies);
+    let speciesId = await connection.query(querySpecies);
```
Comment (byran) [inline central_web_server/handle_database.js]: As per previous Id comment

File: central_web_server/handle_database.js [hunk 8]
```diff
@@ -1,19 +1,137 @@
...
+  }
+  //console.log(groupEntryData);
+  let groupEntryQuery = squel
+    .insert()
+    .into("species_group_entry")
+    .setFieldsRows([].concat.apply([], groupEntryData))
+    .toString()
+  console.log(groupEntryQuery);
+  let groupEntryId = await connection.query(groupEntryQuery);
+  console.log(groupEntryId);
```
Comment (byran) [inline central_web_server/handle_database.js]: Remove this line

File: central_web_server/handle_database.js [hunk 9]
```diff
@@ -1,19 +1,137 @@
...
+      group.species = [];
+    } else {
+      exists = true;
+      group = structureObj[row.species_group_id.toString()]
+    }
+    group.species.push({
+      id: row.species_id,
+      name: row.species_name
+    });
+    console.log(group);
```
Comment (byran) [inline central_web_server/handle_database.js]: Remove this line

**Review by byran — CHANGES_REQUESTED** [review]

**Pull request merged**